src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3269,9 +3269,9 @@ trait Types
       bound.decls enter bsym
       bound
     }
-    def unapply(tp: Type): Option[(TypeName, Type)] = tp match {
```

Comment:
You could always use the boolean extractor pattern:
```scala
def unapply(tp: Type): Boolean = cond(tp) {
  case RefinedType(List(Wildcard Type), scope) => scope.size == 1
}
```
and then 
```scala
else tp match {
  case HasTypeMember() => stopWidening() 
  ...
} 
```

Reply:
Although, why is it only when there's one sym in scope?

Reply:
I did not know about the boolean extractor pattern. Does it have any overhead over the current solution? 

As to why it requires single-item in the Scope, I do not know.

Reply:
I have applied your suggestion. It keeps lines, but I like smaller diffs.